### PR TITLE
Fix batchnorm accuracy by sycl::rsqrt

### LINKED ISSUE #4905: [release/2.14][B60] Four Inductor E2E models regress to fail_accuracy while 2.13 passed
## Summary

On B60, PyTorch release/2.14 XPU nightly shows four new Inductor E2E `fail_accuracy` results. The same cases passed with PyTorch release/2.13.

## Environment

- Hardware: B60
- Container: `intelgpu/ubuntu-26.04-rolling:26.18`
- Target PyTorch: `torch==2.14.0.dev20260805+xpu`
- Target torchvision: `torchvision==0.29.0.dev20260806+xpu`
- Target torchaudio: `torchaudio==2.11.0.dev20260806+xpu`
- Target Triton XPU: `triton-xpu==3.8.0+git1e2d42a0`
- Baseline: PyTorch release/2.13 XPU wheel, same tests passed
- Device: XPU
- Backend: Inductor
- CUDAGraphs: disabled
- Models are not included in pytorch CI or HUD

## Affected Cases

| Suite | Dtype | Mode | Model | Batch size | Result | Key failure |
|---|---|---:|---|---:|---|---|
| `timm_models` | `amp_bf16` | training | `fbnetv3_b` | 8 | `fail_accuracy` | `blocks.0.0.bn1.bias.grad` RMSE exceeds tolerance |
| `torchbench` | `float32` | inference | `detectron2_maskrcnn_r_50_fpn` | 4 | `fail_accuracy` | `pred_masks` / `instances` uint8 mismatch |
| `torchbench` | `float32` | inference | `detectron2_maskrcnn` | 4 | `fail_accuracy` | `pred_masks` / `instances` uint8 mismatch |
| `torchbench` | `bfloat16` | training | `timm_efficientnet` | 4 | `fail_accuracy` | `bn1.bias.grad` RMSE exceeds tolerance |

## Reproducer

```bash
docker run -it -e TZ=Asia/Shanghai \
  --device=/dev/mem --device=/dev/dri --group-add video --privileged \
  --shm-size=8g intelgpu/ubuntu-26.04-rolling:26.18 bash

curl -LsSf https://astral.sh/uv/install.sh | sh
source $HOME/.local/bin/env
uv venv myvenv --python 3.10 --clear
source myvenv/bin/activate
uv pip install pip wheel "setuptools<81"

uv pip install \
  torch==2.14.0.dev20260805+xpu \
  torchaudio==2.11.0.dev20260806+xpu \
  torchvision==0.29.0.dev20260806+xpu \
  triton-xpu==3.8.0+git1e2d42a0 \
  --index-url https://download.pytorch.org/whl/nightly/xpu --pre

pytorch_commit="$(python -c 'import torch; print(torch.version.git_version)')"
git clone https://github.com/pytorch/pytorch
cd pytorch
git checkout ${pytorch_commit}

uv pip install pip pandas psutil scipy requests
uv pip install -r .ci/docker/ci_commit_pins/huggingface-requirements.txt
uv pip install timm

git clone https://github.com/intel/torch-xpu-ops
rsync -avz ./torch-xpu-ops/.ci/benchmarks/ ./benchmarks/dynamo/

git clone https://github.com/pytorch/benchmark tmp-benchmark
cd tmp-benchmark
python install.py --continue_on_fail fbnetv3_b detectron2_maskrcnn_r_50_fpn detectron2_maskrcnn timm_efficientnet
export PYTHONPATH=${PWD}:${PYTHONPATH}
cd ..
```

Representative single-case commands:

```bash
python benchmarks/dynamo/timm_models.py -d xpu --accuracy --amp --amp-dtype bfloat16 \
  --training --backend=inductor --cold-start-latency -n10 \
  --disable-cudagraphs --only fbnetv3_b

python benchmarks/dynamo/torchbench.py -d xpu --accuracy --float32 \
  --inference --backend=inductor --cold-start-latency -n10 \
  --disable-cudagraphs --only detectron2_maskrcnn_r_50_fpn

python benchmarks/dynamo/torchbench.py -d xpu --accuracy --float32 \
  --inference --backend=inductor --cold-start-latency -n10 \
  --disable-cudagraphs --only detectron2_maskrcnn

python benchmarks/dynamo/torchbench.py -d xpu --accuracy --bfloat16 \
  --training --backend=inductor --cold-start-latency -n10 \
  --disable-cudagraphs --only timm_efficientnet
```

## Observed Results

```csv
dev,elapsed,suite,dtype,mode,name,scenario,batch_size,accuracy,calls_captured,unique_graphs,fallbacks_to_eager,graph_breaks,unique_graph_breaks,autograd_captures,autograd_compiles,cudagraph_skips,compilation_latency
xpu,187.811,timm_models,amp_bf16,training,fbnetv3_b,accuracy,8,fail_accuracy,459,2,0,5,4,0,0,0,68.486656
xpu,133.380,torchbench,float32,inference,detectron2_maskrcnn_r_50_fpn,accuracy,4,fail_accuracy,1246,73,4,51,5,0,0,0,69.747165
xpu,133.556,torchbench,float32,inference,detectron2_maskrcnn,accuracy,4,fail_accuracy,1246,73,4,51,5,0,0,0,68.819831
xpu,110.783,torchbench,bfloat16,training,timm_efficientnet,accuracy,4,fail_accuracy,239,1,0,5,4,0,0,0,36.638270
```

Representative failure messages:

```text
fbnetv3_b:
RMSE (res-fp64): 0.01330, (ref-fp64): 0.00232 and shape=torch.Size([16]). res.dtype: torch.float32, multiplier: 3.000000, tol: 0.010000
Accuracy failed for key name blocks.0.0.bn1.bias.grad

detectron2_maskrcnn_r_50_fpn / detectron2_maskrcnn:
Accuracy failed: uint8 tensor did not match
Accuracy failed for key name pred_masks
Accuracy failed for key name instances

timm_efficientnet:
RMSE (res-fp64): 0.00024, (ref-fp64): 0.00004 and shape=torch.Size([32]). res.dtype: torch.bfloat16, multiplier: 3.000000, tol: 0.001000
Accuracy failed for key name bn1.bias.grad
```

## Expected Behavior

These four cases should pass accuracy on B60 with PyTorch release/2.14, matching the PyTorch release/2.13 baseline result.

## Notes

- Two failures are training gradient accuracy mismatches in CNN models.
- Two failures are Detectron2 inference mask/instance mismatches in `uint8` outputs.
- `detectron2_maskrcnn_r_50_fpn` and `detectron2_maskrcnn` show the same captured graph counts and failure keys, so they may share the same underlying regression.

### PR DESCRIPTION
**Pull request by jianyizh**
Align  rsqrt with inductor. Should fix timm_efficientnet accuracy issue in https://github.com/intel/torch-xpu-ops/issues/4905. For detectron2_maskrcnn and detectron2_maskrcnn_r_50_fpn, add them to use_iou_for_bool_masks
After https://github.com/pytorch/pytorch/pull/190206, inductor use rsqrt instead of 1/sqrt
Currently, the compute order of eager and inductor is still different, but changing them will cause UT accuracy fail

### CODE REVIEW

**Comment by jianyizh** [comment]
@torchxpubot review

**Comment by jianyizh** [comment]
@torchxpubot review

**Comment by jianyizh** [comment]
@torchxpubot review

File: .ci/benchmarks/torchbench.yaml [hunk 1]
```diff
@@ -72,8 +72,10 @@ tolerance:
     - sam
     - sam_fast
     - vision_maskrcnn
+    - detectron2_maskrcnn
```
Comment (weishi-deng) [inline .ci/benchmarks/torchbench.yaml:75]: Is this aligned with CUDA's behavior?

**Comment by jianyizh** [comment]
@torchxpubot lint

**Comment by jianyizh** [comment]
@copilot fix lint issue in ci

File: .ci/benchmarks/torchbench.yaml (hunk 1 above)
Comment (jianyizh) [reply .ci/benchmarks/torchbench.yaml:75]: No, but I think it's resonable

**Review by CuiYifeng — APPROVED** [review]
Overall LGTM.

**Review by weishi-deng — APPROVED** [review]

**Comment by jianyizh** [comment]
@torchxpubot ut-check

**Review by chuanqi129 — APPROVED** [review]

**Comment by jianyizh** [comment]
@torchxpubot merge -f

**Comment by jianyizh** [comment]
@torchxpubot merge -f

**Comment by chuanqi129** [comment]
@torchxpubot cherry-pick --onto release/2.14 -c critical
